src/backend/common/queries/dict_converters/converter_base.py

```diff
@@ -0,0 +1,58 @@
+import abc
+from backend.common.queries.types import QueryReturn
+from typing import Callable
+
+
+class ConverterBase(object):
+    @classmethod
+    def convert(cls, thing: QueryReturn, version: int) -> dict:
+        converted_thing = cls._convert(cls._listify(thing), version)
+        if isinstance(thing, list):
+            return cls._listify(converted_thing)
+        else:
+            return cls._delistify(converted_thing)
+
+    @classmethod
+    @abc.abstractmethod
+    def _convert(cls, thing: QueryReturn, version: int) -> Callable:
+        ...
+
+    @classmethod
+    def _listify(cls, thing):
```

Comment:
We may need to have separate type hinted classes for lists?

Or, we'd need to maybe make the generic type here

```
Union[QueryReturn, List[QueryReturn]]
```

Reply:
I think I'd prefer separate classes for list types (since they have their own `QueryReturn` types as well), we'd just have to make the base class aware enough?